sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -434,6 +449,11 @@ case class CreateNamedStruct(children: Seq[Expression]) extends Expression {
   }
 
   override def prettyName: String = "named_struct"
+
+  override def sql: String = getTagValue(FUNC_ALIAS).map { alias =>
```

Comment:
Shouldn't you override prettyName as well? In general this seems to have been solved with either a mixin or at the Expression level.

Reply:
Yup, should be better to do that.